tests/tensor_search/test_search.py

```diff
@@ -54,6 +56,77 @@ def test_each_doc_returned_once(self):
         )
         assert len(search_res['hits']) == 2
 
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '2'}})
+    def test_search_with_excessive_searchable_attributes(self):
+        with self.assertRaises(InvalidArgError):
+            tensor_search.add_documents(
+                config=self.config, index_name=self.index_name_1, docs=[
+                    {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                    {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+                ], auto_refresh=True)
+            tensor_search.search(
+                config=self.config, index_name=self.index_name_1, text="Exact match hehehe",
+                searchable_attributes=["abc", "def", "other field"], return_doc_ids=True
+            )
+
+
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '2'}})
+    def test_search_with_allowable_num_searchable_attributes(self):
+        tensor_search.add_documents(
+            config=self.config, index_name=self.index_name_1, docs=[
+                {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+            ], auto_refresh=True)
+        tensor_search.search(
+            config=self.config, index_name=self.index_name_1, text="Exact match hehehe",
+            searchable_attributes=["other field"], return_doc_ids=True
+        )
+
+    @mock.patch.dict('os.environ', {**os.environ, **{'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': '0'}})
+    def test_search_with_excessive_searchable_attributes(self):
+        with self.assertRaises(InvalidArgError):
+            tensor_search.add_documents(
+                config=self.config, index_name=self.index_name_1, docs=[
+                    {"abc": "Exact match hehehe", "other field": "baaadd", "_id": "5678"},
+                    {"abc": "random text", "other field": "Close match hehehe", "_id": "1234"},
+                ], auto_refresh=True)
+            tensor_search.search(
+                config=self.config, index_name=self.index_name_1, text="Exact match hehehe",
+                searchable_attributes=["other field"], return_doc_ids=True
+            )
+    
+    def test_search_with_searchable_attributes_max_attributes_is_none(self):
+        # Cannot set {'MARQO_MAX_SEARCHABLE_TENSOR_ATTRIBUTES': None} in mock.patch.dict
```

Comment:
i managed to mock environ here: https://github.com/marqo-ai/marqo/blob/mainline/tests/tensor_search/test_add_documents.py#L1099

Reply:
Interesting. Mine produced a unittest error. perhaps either because
1. We're using it as a method decorator (outside even a unittest function)
2. `mock.patch.dict` vs `mock.patch`

Reply:
It is `mock.patch.dict`